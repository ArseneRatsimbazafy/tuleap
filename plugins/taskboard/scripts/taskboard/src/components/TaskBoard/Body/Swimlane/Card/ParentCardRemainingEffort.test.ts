/*
 * Copyright (c) Enalean, 2019 - present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import ParentCardRemainingEffort from "./ParentCardRemainingEffort.vue";
import { Card } from "../../../../../type";

describe("ParentCardRemainingEffort", () => {
    it("displays the remaining effort of the parent card in a badge", () => {
        const wrapper = shallowMount(ParentCardRemainingEffort, {
            propsData: {
                card: {
                    remaining_effort: { value: 666 },
                    color: "lake-placid-blue"
                } as Card
            }
        });
        expect(wrapper.element).toMatchSnapshot();
    });

    it("displays nothing if the parent card has no remaining effort field", () => {
        const wrapper = shallowMount(ParentCardRemainingEffort, {
            propsData: {
                card: {
                    remaining_effort: null,
                    color: "lake-placid-blue"
                } as Card
            }
        });
        expect(wrapper.isEmpty()).toBe(true);
    });

    it("displays nothing if the parent card has no remaining effort value", () => {
        const wrapper = shallowMount(ParentCardRemainingEffort, {
            propsData: {
                card: {
                    remaining_effort: { value: null },
                    color: "lake-placid-blue"
                } as Card
            }
        });
        expect(wrapper.isEmpty()).toBe(true);
    });
});
