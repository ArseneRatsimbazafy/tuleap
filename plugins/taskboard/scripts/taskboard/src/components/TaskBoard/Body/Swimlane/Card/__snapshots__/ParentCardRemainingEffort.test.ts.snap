// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentCardRemainingEffort displays the remaining effort of the parent card in a badge 1`] = `
<span
  class="taskboard-card-remaining-effort tlp-badge-lake-placid-blue taskboard-card-remaining-effort-lake-placid-blue"
>
  
    666
    
  <i
    class="fa fa-long-arrow-right"
  />
   
  <i
    class="fa fa-flag-checkered"
  />
</span>
`;
